lib/DatabaseInfluxDB1x.js

```diff
@@ -61,19 +61,22 @@ class DatabaseInfluxDB1x extends Database{
                         retentionName = row.name;
                     }
                 });
-                callback({"name": retentionName, "time": retentionTime});
+                callback(null, {"name": retentionName, "time": retentionTime});
             }
         });
     }
 
     async applyRetentionPolicyToDB(dbname, retention, callback) {
-        this.getRetentionPolicyForDB(dbname, oldRetention => {
+        this.getRetentionPolicyForDB(dbname, (err, oldRetention) => {
+            //Check, if there was an error
+            if (err)
+                callback(err);
```

Comment:
but I thiunk yoiu wantto "return" if error and not execute the other code or?!

Reply:
Added return back into it. You're right.